### PR TITLE
Select: add default stateful components

### PR DESCRIPTION
**Pull request by sebastiansandqvist**
## Overview
This PR adds a default stateful `<Select>` and `<MediaSelect>` to help cut down on the boilerplate for the most common use cases.

## Additional details
I added many more select demos, so there should now be pretty complete api coverage in those demos.

I also got rid of the `tag` option in the demo ui `<Title>` component since I was just using it to point out whether components were stateful or stateless. Now that we can offer both, the tag might just be confusing. Not 100% sure about this decision, so open to putting it back if it's actually useful.

This is an example of what the tag was:
screen shot 2017-07-12 at 1 56 11 pm

### CODE REVIEW

**Comment by sebastiansandqvist** [comment]
To remove some ambiguity from the examples, I have added code like this:

```
import { StatefulSelect } from '@vhx/quartz-react';

<StatefulSelect options={...} />
```

The reason for that is to make it clear that this is usage of the default stateful select, not the custom stateful select that appears just above it in the examples.

Could use some input on whether that is useful and if that's how we'll be naming the npm module.

**Comment by davegonzalez** [comment]
@sebastiansandqvist I think it's fine to remove the stateful/stateless component. If we find that we, or anyone else, is confused we can look to add it back, or find another way to alleviate the confusion.

**Comment by sebastiansandqvist** [comment]
@davegonzalez Do you mean to remove it from the demo or get rid of the stateful component altogether? That is, we would only provide a stateless one?

I'm open to this, but can imagine even for our own internal use that we'll likely use the default stateful one most of the time.

**Comment by davegonzalez** [comment]
@sebastiansandqvist Sorry for the confusion, I meant it's fine to remove the `Tag` component.

File: .eslintrc [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": "@vhx/eslint-config-vhx"
+  "extends": "@vhx/eslint-config-vhx",
+  "rules": {
+    "jsx-a11y/href-no-hash": 0
```
Comment (sebastiansandqvist) [inline .eslintrc]: I think this rule was added to @vhx/eslint-config-vhx, but it wasn't published to npm. For some reason, this causes a linting error on every file.

Comment (sebastiansandqvist) [inline .eslintrc]: Once the latest version of our eslint config is published, we can remove this one-off rule.

**Pull request merged**